spring-cloud-azure-context/src/main/java/com/microsoft/azure/spring/cloud/context/core/AzureAdmin.java

```diff
@@ -93,4 +94,39 @@ public boolean eventHubConsumerGroupExists(String namespace, String name, String
         return azure.eventHubs().getByName(resourceGroup, namespace, name).listConsumerGroups().stream()
                     .anyMatch(c -> c.equals(group));
     }
+
+    public void createSqlDatabaseIfNotExists(String sqlServerName, String databaseName) {
+        if (!sqlDatabaseExists(sqlServerName, databaseName)) {
+            azure.sqlServers().databases().define(databaseName)
+                 .withExistingSqlServer(resourceGroup, sqlServerName, region).create();
+        }
+    }
+
+    public void createSqlServer(String sqlServerName, String username, String password) {
+        azure.sqlServers().define(sqlServerName).withRegion(region).withExistingResourceGroup(resourceGroup)
+             .withAdministratorLogin(username).withAdministratorPassword(password).create();
+    }
+
+    public SqlServer getOrCreateSqlServer(String sqlServerName, String username, String password) {
+        SqlServer sqlServer = getSqlServer(sqlServerName);
+
+        if (sqlServer != null) {
+            return sqlServer;
+        }
+
+        return azure.sqlServers().define(sqlServerName).withRegion(region).withExistingResourceGroup(resourceGroup)
```

Comment:
Why cannot we call the previous `createSqlServer` method directly without duplication?

Reply:
Will call it.